### ISSUE TITLE
Delegator Contract value doesn't change after adder or subber

### ISSUE THREAD

**Issue by 123oli** [paritytech/ink#1629]
I have uploaded accumulator, adder, subber contract and delegator contract with hash of all those three contracts. Here what I have got:

- Execute `inc` function from adder contract and `read` value from accumulator and got correct values, but delegator contract `read` didn't return correct value(get initial value which is 0).

Add value
<img width="1440" alt="Screenshot 2023-01-30 at 17 19 09" src="https://user-images.githubusercontent.com/42571338/215468184-b1f73cb6-61e4-4134-a79e-1e078b1bee20.png">

Read from accumulator 60 as 50 was already there.
<img width="1440" alt="Screenshot 2023-01-30 at 17 19 57" src="https://user-images.githubusercontent.com/42571338/215468450-dc1893b9-58f6-4b3f-82c1-9b4144c49462.png">

But in the delegator contract when read it return 0.  If It read from accumulator then shouldn't delegator contract return 60 instead of 0?
<img width="1440" alt="Screenshot 2023-01-30 at 17 20 17" src="https://user-images.githubusercontent.com/42571338/215468709-8905d1f0-28f6-473b-90bc-8273c2a31e20.png">

For code-hases I used those highlighted ones.
![Screenshot 2023-01-30 17 40 49](https://user-images.githubusercontent.com/42571338/215470829-05e9719b-d697-48fd-906f-81cd4a20c7b8.png)

Isn't that Issue? Or I am doing something wrong?

**Comment by statictype** [comment]
@123oli how did you access this version of polkadot.js? the live one doesn't seem to have support for `Weightv2`

**Comment by 123oli** [comment]
> @123oli how did you access this version of polkadot.js? the live one doesn't seem to have support for `Weightv2`

Yeah, As I was having issue, I was recommended to use [this version](https://polkadotjs-apps.web.app/#/contracts)

**Comment by ganesh1233456** [comment]
When a smart contract call function of another smart contract, then it can read the state of another contract but modify state is not possible because smart contract operates on sandbox environment hence state of each another contract isn't directly accessible.
